### ISSUE TITLE
Route53 always tries to create

### ISSUE THREAD

**Issue by jzarakas** [capitalone/Particle-Cloud-Framework#88]
This seems to be a boto thing, but not positive. PCF can create an intial record just fine, but does not update on subsequent runs. PCF returns a cannot create, record already exists (don't have access to raw error at the moment, but can dig it up). This error seems to be propagated directly from boto. 

Route53 definition: 
```
route53_defintion = {
    "pcf_name": "route_53",
    "flavor": "route53_record",
    "parents":["ec2_instance:pcf-themis"],
    "aws_resource":{
        "Name":"test.example.com",
        "HostedZoneId":"Z2DM6O3WQ",
        "TTL":30,
        "Type":"A",
        "ResourceRecords":[{"Value":"$inherit$ec2_instance:pcf-themis$PrivateIpAddress"}]
    },
}
```

boto is returning my name as `test.example.com.` When PCF attempts to compare the name, it fails as my domain does not have the trailing `.` Thus, PCF attempts to create a new resource rather than update. 

My fix was to append the trailing `.` when PCF does the name comparison. With this, the record gets updated as expected. 

Happy to submit a PR but it feels dirty. I don't know why the trailing `.` exists as well. 

@bhl5kj helped me debug the issue so he can provide more context if necessary.

**Comment by davidyum** [comment]
hmm yeah I'm thinking something might have changed with boto3 update. Will look into this.

**Comment by ethanwlo** [comment]
Fix will be available in 0.4.1
